### PR TITLE
Adapt StorageExamplesTests

### PR DESCRIPTION
**Pull request by lukasz-golebiewski**
## Overview
Adapts existing tests for use with RSpace which uses the new HotStore under the hood

### JIRA ticket:
https://rchain.atlassian.net/browse/RCHAIN-3254

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

File: rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageExamplesTests.scala [hunk 1]
```diff
@@ -0,0 +1,319 @@
...
+      _ = r9 shouldBe None
+      _ = r10 shouldBe defined
+
+      _ = runK(r10)
+      _ = getK(r10).results shouldBe List(
+        List(carol, carol, carol, carol, alice, alice, alice, bob, bob)
+      )
+
+    } yield (store.changes().map(_.isEmpty shouldBe true))
+  }
```
Comment (odeac) [inline rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageExamplesTests.scala]: I'm feeling a bit lost in so many names.

Can you make this test use fewer channels in order to make the test easier to understand?

Comment (lukasz-golebiewski) [reply rspace/src/test/scala/coop/rchain/rspace/nextgenrspace/StorageExamplesTests.scala]: This isn't the goal of the PR. The aim here is to adapt the existing https://github.com/rchain/rchain/blob/dev/rspace/src/test/scala/coop/rchain/rspace/StorageExamplesTests.scala to work with the recently introduced RSpace which uses `HotStore` under the hood. 

This means that any changes which aren't absolutely necessary for the tests to compile and pass aren't welcome. Simplifying them could be done in a different PR at a later stage

**Pull request merged**